frontend/src/components/FilterDropdown.tsx

```diff
@@ -94,53 +94,65 @@ export const FilterDropdown = (props: FilterDropdownProps) => {
   const [bedBathState, setBedBathState] = useState({
     beds: 1,
     baths: 0.5,
+  });
+  const [bedBathDisplayState, setBedBathDisplayState] = useState({
     bedsDisplay: 1,
     bathsDisplay: 0.5,
     notApplied: true,
   });
   const [searchText, setSearchText] = useState("");
   const [sortIndex, setSortIndex] = useState(0);
   const [availabilityState, setAvailabilityState] = useState({
-    selectedIdx: 0,
     dropdownText: "Available",
   });
+  const [AvailabilityDisplayState, setAvailabilityDisplayState] = useState({
+    selectedIdx: 0,
+  });
   const [priceState, setPriceState] = useState({
     minPrice: -1,
     maxPrice: -1,
+  });
+
+  const [priceDisplayState, setPriceDisplayState] = useState({
     minPriceDisplay: -1,
     maxPriceDisplay: -1,
     notApplied: true,
   });
 
   const applyFilters = () => {
     const filters = {
-      search: searchText ?? undefined,
-      beds: bedBathState.beds,
-      baths: bedBathState.baths,
-      sort: sortIndex,
+      search: searchText ?? "undefined",
+      beds: String(bedBathState.beds),
+      baths: String(bedBathState.baths),
+      sort: String(sortIndex),
       availability: availabilityState.dropdownText,
-      minPrice: priceState.minPrice === -1 ? undefined : priceState.minPrice,
-      maxPrice: priceState.maxPrice === -1 ? undefined : priceState.maxPrice,
+      minPrice: priceState.minPrice === -1 ? "undefined" : String(priceState.minPrice),
+      maxPrice: priceState.maxPrice === -1 ? "undefined" : String(priceState.maxPrice),
```

Comment:
Is `"undefined"` in quotes on purpose?

Reply:
Yes because they are converted to string when sending them as a query parameter so I just wanted to be more clear.